crates/context_aware_config/src/api/config/helpers.rs

```diff
@@ -32,113 +10,14 @@ pub fn apply_prefix_filter_to_config(
-        .clone()
-        .into_iter()
-        .filter(|(key, _)| {
-            prefix_list
-                .iter()
-                .any(|prefix_str| key.starts_with(prefix_str))
-        })
-        .collect();
-
-    for (key, overrides) in &config.overrides {
-        let override_map: Map<String, Value> = overrides.to_owned().into();
-        let filtered_overrides_map: Map<String, Value> = override_map
-            .into_iter()
-            .filter(|(key, _)| filtered_default_config.contains_key(key))
-            .collect();
-
-        if !filtered_overrides_map.is_empty() {
-            filtered_overrides.insert(
-                key.clone(),
-                Cac::<Overrides>::try_from_db(filtered_overrides_map)
-                    .map_err(|err| {
-                        log::error!("filter_config_by_prefix : failed to decode overrides from db with error {}", err);
-                        unexpected_error!(err)
-                    })?
-                    .into_inner(),
-            );
-        }
-    }
-
-    let filtered_context: Vec<Context> = config
-        .contexts
-        .clone()
-        .into_iter()
-        .filter(|context| filtered_overrides.contains_key(&context.override_with_keys[0]))
-        .collect();
-
-    let filtered_config = Config {
-        contexts: filtered_context,
-        overrides: filtered_overrides,
-        default_configs: filtered_default_config,
-    };
-
-    Ok(filtered_config)
-}
-
-pub fn filter_config_by_dimensions(
-    config: &Config,
-    dimension_data: &Map<String, Value>,
-) -> superposition::Result<Config> {
-    let filtered_context = filter_context(&config.contexts, dimension_data);
-    let filtered_overrides: HashMap<String, Overrides> = filtered_context
-        .iter()
-        .flat_map(|ele| {
-            let override_with_key = &ele.override_with_keys[0];
-            config
-                .overrides
-                .get(override_with_key)
-                .map(|value| (override_with_key.to_string(), value.clone()))
-        })
-        .collect();
-
-    let filtered_config = Config {
-        contexts: filtered_context,
-        overrides: filtered_overrides,
-        default_configs: config.default_configs.clone(),
-    };
-
-    Ok(filtered_config)
-}
-
-pub fn get_query_params_map(
-    query_string: &str,
-) -> superposition::Result<Map<String, Value>> {
-    let params =
-        Query::<HashMap<String, String>>::from_query(query_string).map_err(|err| {
-            log::error!("failed to parse query params with err: {}", err);
-            bad_argument!("error getting query params")
-        })?;
-
-    let query_params_map = params
-        .0
-        .into_iter()
-        .map(|(key, value)| {
-            let parsed_value = value
-                .parse::<serde_json::Value>()
-                .unwrap_or_else(|_| json!(value));
-            (key, parsed_value)
-        })
-        .collect::<Map<String, Value>>();
-
-    Ok(query_params_map)
-}
```

Comment:
INFO: Moved to [here](https://github.com/juspay/superposition/pull/264/files#diff-a23f25de1f736bfb011d6a67385bc6ccf43d43208d277669afa9d0db2386d986R139-R158)